### DISCUSSION TITLE
Is it possible to manage the downloading of the tiles via flutter CacheManager ?

### DISCUSSION THREAD

**Discussion by FabriceACohen** [maplibre/flutter-maplibre-gl#387] [Q&A]
Hello,
I am new to maplibre and start to test all its capabilities that seem very nice.
I was wondering : In my app I try to pilot the different network requests to prioritarize the traffic (images, videos) and I wish I could manage map requests to the tile server.
Is there a way to do so ?
Many thanks.

**Comment by m0nac0** [comment]
Sorry, I don't quite understand what you are trying to achieve. Are you trying to set priorities for network requests made by the plugin when requesting map data  compared to other network requests made by your Flutter project (e.g. in case of limited bandwidth)?

**Comment by FabriceACohen** [comment]
Exactly ! I would like to know if the map is downloading, start downloading, stops downloading, even beeing able to pause/restart download in order to priotarize requests between different objects that need bandwith. For exemple in FlutterMap with rasterLayers, there is a TileProvider parameter that allow to set a request controller that provides the data to the map that you can manage (between the network and the map). Thank you.

**Reply by m0nac0** [reply]
This is not currently supported and non-trivial to implement, because the network requests are managed by native code in the MapLibre libraries we are using. There are some options to control the network behavior, but I do not know if we could implement such fine-grained control and it would likely be quite a bit of work.
The only "workaround" I can offer you, is downloading tiles ahead of time (when your users have a good internet connection), then they will be able to use the map even when offline.

**Reply by FabriceACohen** [reply]
Thank you, that is what I was aimed testing (cached areas). But my tested were so so. Do you think there is a way to relay at Flutter level the native network options as they are ?
